AutoRaise.mm

```diff
@@ -475,48 +475,140 @@ CGEventRef eventTapHandler(CGEventTapProxy proxy, CGEventType type, CGEventRef e
 }
 
 #define POLLING_MS 20
+#define VERSION "1.8"
 int main(int argc, const char * argv[]) {
+    printf("\nv%s by sbmpost(c) 2021, usage:\nAutoRaise -delay <1=%dms> [-warpX <0.5> -warpY <0.5> -scale <2.0>]", VERSION, POLLING_MS);
     @autoreleasepool {
-        if (argc >= 3) {
-            NSUserDefaults * standardDefaults = [NSUserDefaults standardUserDefaults];
-            delayCount = abs((int) [standardDefaults integerForKey: @"delay"]);
-            warpMouse = argc != 3 ;
-            if (argc >= 7) {
-                warpX = [standardDefaults floatForKey: @"warpX"];
-                warpY = [standardDefaults floatForKey: @"warpY"];
-                if (argc >= 8) {
-                    cursorScale = [standardDefaults floatForKey: @"scale"];
-                }
+        
+        #define defaultDelay @"delay"
+        #define defaultWarpX @"warpX"
+        #define defaultWarpY @"warpY"
+        #define defaultScale @"scale"
+        
+        // initial values for standardDefaults
+        NSDictionary* standardDefaults = @{defaultDelay:@-1,
+                                           defaultWarpX:@-1.0,
+                                           defaultWarpY:@-1.0,
+                                           defaultScale:@2.0 };
+        
+        // sync standardDefaults with command-line arguments
+        [[NSUserDefaults standardUserDefaults] registerDefaults:standardDefaults];
+        
+        // put initial values to our variables
+        delayCount = (int) [[NSUserDefaults standardUserDefaults] integerForKey:defaultDelay];
+        warpX = (float) [[NSUserDefaults standardUserDefaults] floatForKey:defaultWarpX];
+        warpY = (float) [[NSUserDefaults standardUserDefaults] floatForKey:defaultWarpY];
+        cursorScale = (float) [[NSUserDefaults standardUserDefaults] floatForKey:defaultScale];
+
+        // if there are any arguments, we would use only them
+        if (argc > 1) {
+            // but if -delay argument not valid, we would exit
+            if (not (delayCount > 0)) {
+                fprintf(stderr, "No valid arguments\nRun without any arguments to use default settings");
+                return 0;
             }
+            // check is there valid warps
+            if (warpX >= 0 and warpX <= 1 and warpY >= 0 and warpY <= 1) { warpMouse = true; }
```

Comment:
So you actually chose to implement validation checks. That is cool, but also makes the code quite a bit more complex. So I myself chose to not do any validation checks at all (and instead print out clearly to the command line what settings are being applied).

Reply:
it is not real validation.
this condition only for detecting warpMouse status.
I wanted do it  instead checking arguments quantity - otherwise false positive 'true' value possible sometimes.